src/extension/content-script/batteries/YouTubeVideo.ts

```diff
@@ -62,6 +66,24 @@ const battery = async (): Promise<void> => {
   ]);
 };
 
+const battery = async (): Promise<void> => {
+  setData();
+  const videoDescription = document.querySelector(
+    "div#bottom-row.style-scope.ytd-watch-metadata"
+  );
+  if (!videoDescription) return;
+
+  const observer = new MutationObserver((mutations) => {
```

Comment:
let's debounce this function to not call it too often

e.g. let's debounce it for 200ms or something. @reneaaron what do you think?

https://underscorejs.org/docs/modules/debounce.html

Reply:
I think we can just return if the URL didn't change, I can't see a reason why you would try to fetch this info for the same URL more than once. :thinking: